airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateConverter.java

```diff
@@ -0,0 +1,272 @@
+/*
```

Comment:
This file was moved from the airbyte-server module to airbyte-config, which is why the diff is entirely green. The actual diff is mainly the two new public methods, `toApi` and `toClient`

Reply:
This is what caused `config-model` to depend on `airbyte-api`? I think that's a weird dependency to have. Why should model declarations need to depend on the API? That doesn't sound right. Would it make sense to just move this into its own model?

Reply:
Yeah, this dependency is weird, @davinchia and I were looking at this and trying to pick a sensible spot but we might need to go back to the drawing board.

@cgardens when you said maybe we should just move this into its own model, did you mean _module_? I was thinking we might want to define a new module like `airbyte-converters` that can provide helper methods for converting between client-side models, server-side models, and internal config models.

Dependency management is definitely not my wheelhouse so I'm open to suggestions here!

Reply:
I think having an `airbyte-converters` module may come with a pretty big downside, we end up with one module that handles all conversions, which means that this modules end up depending on all data generated objects.
Trying to untangle this, my suggestion would be to leave the converter you're adding in the `airbyte-workers` rather than introduce unwanted dependencies by consolidating it. I think it remains reasonable for now and saves some heavy lifting. Also, with my suggestion of moving some logic to the handler, the airbyte workers should only need the `toClient` conversion.

Reply:
yeah. own module.

Reply:
discussion should be continued at child PR here: https://github.com/airbytehq/airbyte/pull/15902/files#r953189950